### PR TITLE
Implement the `$cond` operator in the aggregate pipeline. For #349.

### PR DESCRIPTION
**Pull request by pcorpet**

### CODE REVIEW

File: mongomock/collection.py [hunk 1]
```diff
@@ -1581,11 +1581,23 @@ def _handle_conditional_operator(operator, values, doc_dict):
                 except KeyError:
                     return fallback
                 return out_value if out_value is not None else fallback
-            else:
-                raise NotImplementedError(
-                    "Although '%s' is a valid date operator for the "
-                    "aggregation pipeline, it is currently not implemented "
-                    " in Mongomock." % operator)
+            if operator == '$cond':
```
Comment (jheden) [inline mongomock/collection.py]: To be in line with how the other types of operators, and the stages of the aggregation pipeline, are handled, I'd suggest using `elif` here (and maybe keeping the `else` for `NotImplementedError`, even though it is the very last statement in the method).

**Review by jheden — APPROVED** [review]
Nicely done!

**Comment by jheden** [comment]
Again, approving despite a failing test, due to the test failing for unrelated circumstances.

File: mongomock/collection.py (hunk 1 above)
Comment (pcorpet) [reply mongomock/collection.py]: Done.

**Review by jheden — APPROVED** [review]

**Pull request merged**